src/main/kotlin/com/depromeet/team3/wishlist/service/WishPersistenceService.kt

```
@@ -0,0 +1,29 @@
+// 영속화만 별도 빈으로 분리. 같은 빈에서 호출하면 Spring AOP proxy 를
+// 거치지 않아 @Transactional 가 무력화되기 때문이다.
+@Service
+class WishPersistenceService(
+    private val wishRepository: WishRepository,
+) {
+    @Transactional
+    fun persist(guestId: UUID, product: Product): WishRegisterResult =
+        try {
+            val wish = wishRepository.save(Wish(guestId = guestId, product = product))
+            WishRegisterResult(wish = wish)
+        } catch (e: DataIntegrityViolationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)